syntax_and_semantics/cross-compilation.md

```diff
@@ -2,33 +2,60 @@
 
 ```bash
 crystal build your_program.cr --cross-compile --target "x86_64-unknown-linux-gnu"
 ```
 
-This will generate a `.o` ([Object file](http://en.wikipedia.org/wiki/Object_file)) and will print a line with a command to execute on the system we are trying to cross-compile to. For example:
+This will generate a `.o` ([Object file](http://en.wikipedia.org/wiki/Object_file)) and will print a line with a command to execute on the target system. For example:
 
 ```bash
-cc your_program.o -o your_program -lpcre -lrt -lm -lgc -lunwind
+cc your_program.o -o your_program -lpcre -lgc /local/crystal/src/ext/libcrystal.a
 ```
 
-You must copy this `.o` file to that system and execute those commands. Once you do this the executable will be available in that target system.
+The command above can be executed on the target system, once the `.o` file has been copied over.  The reference to `libcrystal.a` may need to be adjusted to match the location that `libcrystal` was compiled to, as per the preparation step.
 
-This procedure is usually done with the compiler itself to port it to new platforms where a compiler is not yet available. Because in order to compile a Crystal compiler we need an older Crystal compiler, the only two ways to generate a compiler for a system where there isn't a compiler yet are:
-* We checkout the latest version of the compiler written in Ruby, and from that compiler we compile the next versions until the current one.
-* We create a `.o` file in the target system and from that file we create a compiler.
+This procedure is usually done with the compiler itself to port it to new platforms where a compiler is not yet available. Because in order to compile a Crystal compiler, an older Crystal compiler is required, the only two ways to generate a compiler for a system where there is no compiler yet are:
+* Checkout the latest version of the compiler written in Ruby, then use that compiler to compile all subsequent versions of the compiler until the current version is reached.
+* Create a `.o` file for the target system, and use it to create a compiler.
 
 The first alternative is long and cumbersome, while the second one is much easier.
 
-Cross-compiling can be done for other executables, but its main target is the compiler. If Crystal isn't available in some system you can try cross-compiling it there.
+Cross-compiling can be done for other executables, but its main target is the compiler. If Crystal is not available in some system, it can theoretically be cross-compiled for that system.
+
+
+## Troubleshooting libraries
+
+The cross-compilation step will generate a command with numerous flags, many of these refer to the libraries that need to be included when compiling a program.
+If the libraries are not present on the target system, the compilation will fail and complain about the respective flags, which are not always descriptive.
+
+The following command may be helpful:
+
+```bash
+ld -lgc --verbose
```

Comment:
This command prints a huge load of stuff that isn't very useful, perhaps a better method is required.

Reply:
It does, but the volume doesn't matter that much right now, the useful output is right at the end, and I don't know of a better alternative, happy to change if you have any suggestions?

Reply:
Well personally I think that having the paths doesn't really help you: you can either search for packages that provide `libfoo.so`, or you have to google `libfoo`. This command only really tells you the linker path, which is knowledge contained in the package already.

Reply:
You say this but I would suggest you have so much integral knowledge by now it's hard to judge.  I had no idea what any of the flags were, seeing the paths was the only thing that allowed me to work out what libraries I needed to install.